hbase-server/src/test/java/org/apache/hadoop/hbase/client/TestFromClientSide.java

```diff
@@ -1542,230 +1564,232 @@ public void testVersionLimits() throws Exception {
-    assertTrue("Expected 9 keys but received " + result.size(),
-        result.size() == 9);
+      get = new Get(ROW);
+      get.setMaxVersions(Integer.MAX_VALUE);
+      result = ht.get(get);
+      assertTrue("Expected 9 keys but received " + result.size(),
+          result.size() == 9);
 
-    scan = new Scan(ROW);
-    scan.setMaxVersions(Integer.MAX_VALUE);
-    scan.addFamily(FAMILIES[0]);
-    scan.addFamily(FAMILIES[1]);
-    scan.addFamily(FAMILIES[2]);
-    result = getSingleScanResult(ht, scan);
-    assertTrue("Expected 9 keys but received " + result.size(),
-        result.size() == 9);
+      get = new Get(ROW);
+      get.addFamily(FAMILIES[0]);
+      get.addFamily(FAMILIES[1]);
+      get.addFamily(FAMILIES[2]);
+      get.setMaxVersions(Integer.MAX_VALUE);
+      result = ht.get(get);
+      assertTrue("Expected 9 keys but received " + result.size(),
+          result.size() == 9);
 
-    scan = new Scan(ROW);
-    scan.setMaxVersions(Integer.MAX_VALUE);
-    scan.addColumn(FAMILIES[0], QUALIFIER);
-    scan.addColumn(FAMILIES[1], QUALIFIER);
-    scan.addColumn(FAMILIES[2], QUALIFIER);
-    result = getSingleScanResult(ht, scan);
-    assertTrue("Expected 9 keys but received " + result.size(),
-        result.size() == 9);
+      get = new Get(ROW);
+      get.addColumn(FAMILIES[0], QUALIFIER);
+      get.addColumn(FAMILIES[1], QUALIFIER);
+      get.addColumn(FAMILIES[2], QUALIFIER);
+      get.setMaxVersions(Integer.MAX_VALUE);
+      result = ht.get(get);
+      assertTrue("Expected 9 keys but received " + result.size(),
+          result.size() == 9);
 
+      scan = new Scan(ROW);
+      scan.setMaxVersions(Integer.MAX_VALUE);
+      result = getSingleScanResult(ht, scan);
+      assertTrue("Expected 9 keys but received " + result.size(),
+              result.size() == 9);
+
+      scan = new Scan(ROW);
+      scan.setMaxVersions(Integer.MAX_VALUE);
+      scan.addFamily(FAMILIES[0]);
+      scan.addFamily(FAMILIES[1]);
+      scan.addFamily(FAMILIES[2]);
+      result = getSingleScanResult(ht, scan);
+      assertTrue("Expected 9 keys but received " + result.size(),
+              result.size() == 9);
+
+      scan = new Scan(ROW);
+      scan.setMaxVersions(Integer.MAX_VALUE);
+      scan.addColumn(FAMILIES[0], QUALIFIER);
+      scan.addColumn(FAMILIES[1], QUALIFIER);
+      scan.addColumn(FAMILIES[2], QUALIFIER);
+      result = getSingleScanResult(ht, scan);
+      assertTrue("Expected 9 keys but received " + result.size(),
+              result.size() == 9);
+    }
   }
 
   @Test
   public void testDeleteFamilyVersion() throws Exception {
-    Admin admin = TEST_UTIL.getAdmin();
-    final TableName tableName = TableName.valueOf(name.getMethodName());
-
-    byte [][] QUALIFIERS = makeNAscii(QUALIFIER, 1);
-    byte [][] VALUES = makeN(VALUE, 5);
-    long [] ts = {1000, 2000, 3000, 4000, 5000};
-
-    Table ht = TEST_UTIL.createTable(tableName, FAMILY, 5);
+    try (Admin admin = TEST_UTIL.getAdmin()) {
```

Comment:
If the TEST_UTIL is not created per test, but is setup once only for all tests, then indeed, closing the Admin made by TEST_UTIL will mess up subsequent attempts at using admin.

Reply:
@petersomogyi Sorry I missed that TEST_UTIL actually stores and maintains the Admin. It should not be closed from tests, but from your comment I cannot decide whether I change it here or just leave, because it's a backport.

Reply:
I think we can have this change like this to be consistent between master and branch-2 and fix the Admin closes in a different ticket since that is required on master as well.